src/libostree/ostree-gpg-verify-result.c

```diff
@@ -773,8 +769,55 @@ ostree_gpg_verify_result_require_valid_signature (OstreeGpgVerifyResult *result,
 
   if (ostree_gpg_verify_result_count_valid (result) == 0)
     {
-      g_set_error (error, OSTREE_GPG_ERROR, OSTREE_GPG_ERROR_MISSING_KEY,
-                   "GPG signatures found, but none are in trusted keyring");
+      /*
+       * Join the description of each failed signature for the error message.
+       * Only one error code can be returned, so if there was more than one
+       * signature, use the error of the last one under the assumption that
+       * it's the most recent and hopefully most likely to be made with a
+       * valid key.
```

Comment:
Maybe we should add the signature errors in reverse order too, following that logic?  

That could clean up the code here a bit too because we could do

```
if (nsigs == 0)
  g_string_append (buffer, "No GPG signatures found");
else 
  {
    /* current i == nsigs - 1 code */

   for (guint i = nsigs - 1; i>= 0; i--)
        ostree_gpg_verify_result_describe_variant (info, buffer, "", ...)
  }
```

or so?  But, it's also fine as is.

Reply:
I think that makes sense. Then the error code would then match the first message instead of the last, but I don't think it matters much.

Reply:
Heh, I just wasted 20 minutes chasing an assertion because a `guint` is always `>= 0`.